src/utils/mod.rs

```diff
@@ -1,20 +1,24 @@
-use crate::cell::point::Point;
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct Point {
+    pub x: u16,
+    pub y: u16,
+}
 
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct Multipolygon {
     pub outer: Vec<Vec<Point>>,
-    pub inner: Vec<Vec<Point>>
+    pub inner: Vec<Vec<Point>>,
 }
 
 impl Multipolygon {
-    pub fn covers(&self, point: &Point) -> bool {
+    pub fn covers(&self, point: Point) -> bool {
         let mut insides = 0;
-        for area in self.outer.iter() {
+        for area in &self.outer {
```

Comment:
Why does this work?

Reply:
https://stackoverflow.com/questions/66908520/rust-iterating-over-iter-or-vector-itself

Reply:
clippy told me